scipy/integrate/tests/test_quadrature.py

```diff
@@ -587,17 +598,20 @@ def test_nd(self, axis, x_ndim, x_len, i_ndim, dx):
         ("`initial` must either be a scalar or...", dict(initial=np.arange(5))),
         ("`dx` must either be a scalar or...", dict(x=None, dx=np.arange(5))),
     ])
-    def test_simpson_exceptions(self, message, kwarg_update):
-        kwargs0 = dict(y=self.y0, x=self.x0, dx=None, initial=None, axis=-1)
+    def test_simpson_exceptions(self, message, kwarg_update, xp):
+        kwargs0 = dict(y=xp.asarray(self.y0), x=xp.asarray(self.x0), dx=None,
+                       initial=None, axis=-1)
+        kwarg_update = {k: xp.asarray(v) if isinstance(v, list) else v
+                        for k, v in kwarg_update.items()}
         with pytest.raises(ValueError, match=message):
             cumulative_simpson(**dict(kwargs0, **kwarg_update))
 
-    def test_special_cases(self):
+    def test_special_cases(self, xp):
         # Test special cases not checked elsewhere
         rng = np.random.default_rng(82456839535679456794)
-        y = rng.random(size=10)
-        res = cumulative_simpson(y, dx=0)
-        assert_equal(res, 0)
+        y = xp.asarray(rng.random(size=10))
+        res = cumulative_simpson(y, dx=0.)
+        xp_assert_equal(res, 0*res)
```

Comment:
Maybe a bit pedantic, but do we really need to include the `actual` value in the `expected` value as well? This seems undesirable on some levels for avoiding weird issues, and also just feels a bit like cross-contamination into the expected value (yeah it is multiplied by 0, but weird things can happen...).